Codecademy/gamut
# feat(MobileBookmarks): Add support for mobile screen window pane

### Overview

This PR:
- fixes sync issue between desktop / mobile notifications (see screenshots below - on prod they currently operate independently since the desktop and mobile are on separate subcomponents that don't share state)
![notifications_bug](https://user-images.githubusercontent.com/34633964/184019485-1aacc372-66a2-45ee-9073-bf6e0080c5b4.png)
- introduces global toggle state between notifications and bookmarks & adds mobile bookmark content pane [video demo](https://www.loom.com/share/39446904a95a48ce94a9d78fccb00ef3)
- fixes anon mobile header button vertical alignment (see screenshots below)
![mobile_align_bug](https://user-images.githubusercontent.com/34633964/184019816-3b46a27c-de5f-465a-a7bf-8a2f96d46b04.png)
![mobile_align_fix](https://user-images.githubusercontent.com/34633964/184019847-45ad82c8-1ca2-4adc-ae9a-7548f14420d5.png)

- adds a minified codecademy logo for `free users` on screens 1200 - 1260px to give more space for app header items [video demo](https://www.loom.com/share/85fce6a0fbf84ddf9a610de250a94f93)


Note: None of these are breaking changes.  We have a ticket ([REACH-1977](https://codecademy.atlassian.net/browse/REACH-1977)) to come back and remove the previous `renderBookmarks` prop related code from gamut once we switch over to using the new `crossDeviceBookmarkParts` prop in portal/monolith.

### PR Checklist

- [x] Related to designs: https://www.figma.com/file/5rkfS3uWZsFNTsx2GmlDYA/Save-Feature-MVP?node-id=462%3A31401
- [x] Related to JIRA ticket: REACH-1972

#### Testing Instructions
NOTE: If testing locally, make sure to have the `bookmarks` feature flag on for your test user.
Visit the [portal-app PR](https://github.com/codecademy-engineering/portal-app/pull/2367) with preview ENV to try out these changes. Use the instructions below when testing the portal-app preview ENV:

1. As an **anonymous** user, visit `/articles`.
2. Click on bookmarks button.
3. Confirm the popover opens and you see the anonymous view copy as well as the `Log In` and `Sign Up` buttons in the menu.
4. Click on the bookmarks button again. Confirm that this action toggles the popover.
5. Check the breakpoint from medium and below.
6. Confirm that you see the pane with anonymous copy. Confirm the pane pushes down content.
7. Confirm the `Log In` and `Sign Up` buttons in the global nav are vertically aligned with the hamburger menu button.
8. Log in as a **free** user. Visit `/articles`.
9. Click on the notifications bell icon.
10. Confirm that notifications swaps from a popover on >large to a pane (that pushes down content) on <medium breakpoints.
11. At the large breakpoint, click on the bookmarks button.
12. Confirm that the notifications popover closes and you see the bookmarks popover.
13. Become a `Pro User (admin user works too)` and check the breakpoint between 1260px and 1200px and confirm you do NOT see a mini CC icon in the top left of your screen.
14. Become a `Free User` (might be easier to do locally)
15. Check the breakpoint between 1260px and 1200px and confirm you DO see a mini CC icon in the top left of your screen.
16. Check the breakpoints outside this 1200-1260 range and confirm the logo is back to normal.



[Don't make me tap the sign.](https://i.imgur.com/sy93D9I.png)

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/portal-app/pull/2367)  | [Portal Env](https://tengu.codecademy.com/catalog?PR_ENV=portal-app-pr-2367)   |
| Monolith | [Monolith Link](https://github.com/codecademy-engineering/Codecademy/pull/32016) | [Monolith ENV](https://pr-32016-monolith.dev-eks.codecademy.com/) |
